main.py

```diff
@@ -115,26 +115,30 @@ def process(data, factors):
             average[factor_set][surface] = np.average(np.array(average[factor_set][surface]), axis=0)
     return average
 
-def sort_directions(data):
+def choose_direction(data, average):
     user_surfaces = []
     for direction_name, surface_arrays in data.items():
-        user_surfaces.append((direction_name, len(surface_arrays.keys()),surface_arrays.keys()))
-    return sorted(user_surfaces, key=lambda x: x[1], reverse=True)
-
+        user_surfaces.append((direction_name, len(surface_arrays.keys())))
+    #get the number of surfaces stored in average as keys
+    user_surfaces = sorted(user_surfaces, key=lambda x: x[1], reverse=True)
+    chosen_surface_num = user_surfaces[0][1]
+    average_surface_num = len(average[next(iter(average))].keys())
+    #I am not sure if this is really necessary to check if they are equal
+    if average_surface_num == chosen_surface_num:
```

Comment:
@harrywhoforks What to do if the statement is False?

Reply:
At first I was thinking:
1. dictionary "average" should accquire all user_surfaces
2. dictionary "data" stores the info about what user_surfaces are in each csv file
3. so if the number of user_surfaces in a csv == number of user_surfaces in "average", then that csv file is the chosen one.

But then i realize I have sorted it to get the csv file with the max user_surfaces already, which is supposed to == number of user_surfaces in "average".....lol

Now I think it might be more appropriate to check exactly all names of user_surfaces in each csv file against all names of user_surfaces in average. Let me try to revise it.

Reply:
Outdated discussion. Resolving.